qiskit/circuit/measure.py

```diff
@@ -59,3 +61,53 @@ def measure(self, qubit, cbit):
 
 
 QuantumCircuit.measure = measure
+
+
+class PauliMeasure(Measure):
+    """Perform a measurement with preceding basis change operations."""
+
+    def __init__(self, basis='z'):
+        """Create a new basis transformation measurement.
+
+        Args:
+            basis (str): The target measurement basis, can be 'x', 'y' or 'z'.
+
+        Raises:
+            ValueError: If an unsupported basis is specified.
+        """
+        super().__init__()
+
+        if basis.lower() == 'x':
+            self.name = 'x_measure'
+            from qiskit.extensions.standard import HGate
+            basis_transformation = [HGate()]
+        elif basis.lower() == 'y':
+            self.name = 'y_measure'
+            from qiskit.extensions.standard import RXGate
+            basis_transformation = [RXGate(pi / 2)]
```

Comment:
You should use two gate transformation here of `SdgGate` and `HGate` here so this can still be a Clifford circuit to run on Stabilizer simulator.

Reply:
Good point, I'll change it to `[SdgGate, HGate, SdgGate]` 🙂